src/tasks/processBulkUpload.ts

```diff
@@ -14,4 +84,16 @@ export const processBulkUpload = async (
     );
   }
   helpers.logger.debug(`Started processBulkUpload Job for Bulk Upload ID ${payload.bulkUploadId}`);
+  const bulkUpload = await loadBulkUpload(payload.bulkUploadId);
+  let bulkUploadFile: FileResult;
+  try {
+    await updateBulkUploadStatus(bulkUpload.id, BulkUploadStatus.IN_PROGRESS);
+    bulkUploadFile = await downloadS3ObjectToTemporaryStorage(bulkUpload.sourceKey);
+  } catch (error) {
+    helpers.logger.info('Bulk upload is being marked as failed', { error });
+    await updateBulkUploadStatus(bulkUpload.id, BulkUploadStatus.FAILED);
+    return;
+  }
+  await bulkUploadFile.cleanup();
+  await updateBulkUploadStatus(bulkUpload.id, BulkUploadStatus.COMPLETED);
```

Comment:
Cleanup might fail and I should add a try / catch to that.

It would not make sense to retry the job though on a cleanup failure -- the file being cleaned is a temporary file being created with a random name every time a job processes, so the second attempt to process would (A) still not clean up and (B) would result in double insertion of proposals in the event that the failed cleanup happened in the end.